### PR TITLE
new icon: hardhat (original)

### LINKED ISSUE #1355: [ICON REQUEST]: Hardhat icon
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### About the icon

Hardhat is a popular Ethereum development environment 

### Links and sources

Link to official website : **https://hardhat.org/**
Link to official Documentation : **https://hardhat.org/docs**
Official Logo link : **https://seeklogo.com/vector-logo/426726/hardhat**

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by knight-95**
## Double check these details before you open a PR

- [x] PR does not match another non-stale PR currently opened
- [x] PR name matches the format *new icon: <i>Icon name</i> (<i>versions separated by comma</i>)*. More details [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#overview)
- [x] PR's base is the `develop` branch.
- [x] Your icons are inside a folder as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#organizational-guidelines)
- [x] SVG matches the standards laid out [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#svgStandards)
- [x] A new object is added in the `devicon.json` file as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#-updating-the-deviconjson-)

**This PR closes NONE**

This PR closes #1355 #1356 

## Link to prove your SVG is correct and up-to-date.

https://hardhat.org/
https://hardhat.org/docs

### CODE REVIEW

File: devicon.json [hunk 1]
```diff
@@ -2158,6 +2158,24 @@
         "color": "#000000",
         "aliases": []
     },
+    {
```
Comment (Snailedlt) [inline devicon.json:2161]: altnames needs to be added, see [updating the devicon.json](https://github.com/devicons/devicon/wiki/Updating-the-%60devicon.json%60)

File: icons/hardhat/hardhat-original.svg [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="256px" height="177px" viewBox="0 0 256 177" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" preserveAspectRatio="xMidYMid">
```
Comment (Snailedlt) [inline icons/hardhat/hardhat-original.svg:2]: Viewbox needs to be 0 0 128 128.
See [SVG STANDARDS](https://github.com/devicons/devicon/wiki/SVG-Standards) and [Recommended resources and tools](https://github.com/devicons/devicon/wiki/Recommended-resources-and-tools) for more info

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Some minor changes that need to be done

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "devicon",
-  "version": "2.15.0",
+  "version": "2.15.1",
```
[the remark this answers is not in the document]
Comment (knight-95) [reply package.json]: I have it updated in the new pull request.

File: devicon.json [hunk 4]
```diff
@@ -2158,6 +2158,28 @@
...
+        "name": "hardhat",
+	"altnames": [
+		"hardhat",
```
Comment (Snailedlt) [inline devicon.json:2164]: No need to add the same name as an altname
```suggestion
```

File: package-lock.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "packages": {
     "": {
       "name": "devicon",
-      "version": "2.14.0",
+      "version": "2.15.1",
```
Comment (Snailedlt) [inline package-lock.json:9]: Still no need to change this :)

**Review by Snailedlt — CHANGES_REQUESTED** [review]

**Review by knight-95 — COMMENTED** [review]
Can you please check now

**Review by knight-95 — COMMENTED** [review]
Updated Viewbox to  0 0 128 128.

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Devicon.json looks good now

Only 3 things left now:
1: remove changes to package.json (see suggestion below)
2: Center and resize icon to fit inside the viewbox
3: Provide a unicolored plain and/or line icon that can be used for fonts. An easy way to do this is to just set all the yellow paths to the same color, and the other paths as transparent

File: icons/hardhat/hardhat-original.svg (hunk 2 above)
Comment (knight-95) [reply icons/hardhat/hardhat-original.svg]: Sorry my fault, now it's centered..

File: icons/hardhat/hardhat-original.svg [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (knight-95) [reply icons/hardhat/hardhat-original.svg:1]: Sorry my bad, have updated the same

**Review by knight-95 — COMMENTED** [review]
Centered the svg icon using recommended tools

File: devicon.json [hunk 7]
```diff
@@ -2158,6 +2158,27 @@
...
+        "versions": {
+            "svg": [
+                "original"
+            ],
+            "font": [
+                "original"
+            ]
+        },
+        "color": "#fef04c",
+        "aliases": []
```
Comment (Panquesito7) [inline devicon.json:2180]:
```suggestion
        "aliases": [
            {
                "base": "original",
                "alias": "plain"
            },
        ]
```


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inefficient

File: icons/hardhat/hardhat-original.svg [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
```
Comment (Snailedlt) [inline icons/hardhat/hardhat-original.svg:1]: Now the icon is stretched :sweat_smile:

**Review by Snailedlt — CHANGES_REQUESTED** [review]

**Review by knight-95 — COMMENTED** [review]
Added the suggested aliases

**Review by knight-95 — COMMENTED** [review]
@Snailedlt 128x128 will make the picture bit unsized 😅, but it works fine...

**Comment by Snailedlt** [comment]
> @Snailedlt 128x128 will make the picture bit unsized sweat_smile, but it works fine...

Try resizing without stretching it. It doesn't have to stretch to fit, just needs to be centered and fit either vertically or horizontally.

I would try and resize the original icon again if I were you :)

**Review by knight-95 — COMMENTED** [review]
@Snailedlt 
Changed svg viewBox="0 0 128 128" According to standards given in documentation
Also the icon now looks good :)

**Comment by Snailedlt** [comment]
> @Snailedlt Changed svg viewBox="0 0 128 128" According to standards given in documentation Also the icon now looks good :)

Nice, looks better now.
Only thing missing is to resize so it fits horizontally and center it. You can use one of the recommended tools in the wiki to do that.

Oh and it would be great if you provided a plain icon where the eth logo is transparent and the rest of the icon is the same color :)

**Comment by BenSouchet** [comment]
By the way on the hardhat website there is a wordmark version of the logo:
https://hardhat.org/_next/static/media/hardhat-logo-dark.484eb916.svg
and / or 
https://hardhat.org/_next/static/media/hardhat-logo.5c5f687b.svg

**Comment by Snailedlt** [comment]
@knight-95 Can you add the wordmark icon suggested by @BenSouchet too? 💯

File: icons/hardhat/hardhat-original.svg (hunk 8 above)
Comment (Snailedlt) [reply icons/hardhat/hardhat-original.svg:1]: @knight-95 still stretched 🙈

**Comment by Snailedlt** [comment]
Superseeded by #1565

**Pull request closed without merging**